### PR TITLE
STY: use pytest.raises context syntax (reshape)

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref #24332

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -1087,7 +1087,7 @@ def _validate(self, validate):
         elif validate in ["one_to_many", "1:m"]:
             if not left_unique:
                 raise MergeError("Merge keys are not unique in left dataset;"
-                                 "not a one-to-many merge")
+                                 " not a one-to-many merge")
```
Comment (simonjayhawkins) [inline pandas/core/reshape/merge.py]: note that error message in main code has been changed

File: pandas/tests/reshape/merge/test_join.py [hunk 2]
```diff
@@ -195,38 +195,47 @@ def test_join_on(self):
...
     def test_join_on_fails_with_different_left_index(self):
-        with pytest.raises(ValueError):
-            df = DataFrame({'a': np.random.choice(['m', 'f'], size=3),
-                            'b': np.random.randn(3)},
-                           index=tm.makeCustomIndex(10, 2))
-            df2 = DataFrame({'a': np.random.choice(['m', 'f'], size=10),
-                             'b': np.random.randn(10)})
+        df = DataFrame({'a': np.random.choice(['m', 'f'], size=3),
+                        'b': np.random.randn(3)},
+                       index=tm.makeCustomIndex(3, 2))
```
Comment (simonjayhawkins) [inline pandas/tests/reshape/merge/test_join.py]: note that this test was failing on the dataframe constructor and not reaching the merge operation. rectified.

File: pandas/tests/reshape/test_pivot.py [hunk 3]
```diff
@@ -742,24 +742,27 @@ def test_margins_no_values_two_row_two_cols(self):
             index=['A', 'B'], columns=['C', 'D'], aggfunc=len, margins=True)
         assert result.All.tolist() == [3.0, 1.0, 4.0, 3.0, 11.0]
 
-    def test_pivot_table_with_margins_set_margin_name(self):
+    @pytest.mark.parametrize(
+        'margin_name', ['foo', 'one', 666, None, ['a', 'b']])
+    def test_pivot_table_with_margins_set_margin_name(self, margin_name):
```
Comment (simonjayhawkins) [inline pandas/tests/reshape/test_pivot.py]: this test parametrized only to the extent to simplify expected error message construction.

**Comment by jreback** [comment]
this broke a 3.6 build with older numpy; which we use when building wheels: https://travis-ci.org/MacPython/pandas-wheels/jobs/481944243

@simonjayhawkins can you do a followup to fix? not easy to test this.

**Pull request merged**